### ISSUE TITLE
[Typo] 09.Templates_I.pdf

### ISSUE THREAD

**Issue by jakeheke75** [federico-busato/Modern-CPP-Programming#39]
Hello Federico,
I have a doubt about page 10. This specific slide is explaining the difference between Implicit and Explicit Template Instantiation.
The last row is showing instruction `template f<int>(int);`
But the explicit template instantiation shouldn't be for instance `f<int>(3);`?
Because the instruction `template f<int>(int);` it seems a template specialization, something like:
```cpp
template<>
void f<int>(int); 
```
I apologize in advance if I misunderstood the meaning of the slide, but I am the perfect "dumb user" since I am learning C++ from your course without any previous experience :smile: 
Thanks,
Gianmario
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/93e0e9e7-cfaa-4909-9324-8fd6a7208728)

**Comment by federico-busato** [comment]
Hi @jakeheke75, I appreciated the question. The line that you underlined contains a type (missing `void`), but the concept is correct. You can explicitly instantiate a template without passing actual parameters. Here a full example https://godbolt.org/z/Gqa8c4WPE. This concept is taken over in `11.Translation Units I - ODR - Function Template` lecture.
Btw, is Gianmario your name, right? Are you from Italy too?

**Comment by federico-busato** [comment]
I updated the slide and added a new one to better show the concept. I also realized that the comment on `f<short>` was wrong

**Comment by jakeheke75** [comment]
Hello Federico, thank you for the update! 
Now it's completely clear and I can also see the difference between implicit and explicit template instantiation.
I tried also to compile the code myself, I am getting a warning message at page 11 because an `int` function is not returning any value. Maybe it should be `void g()`?
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/02fdcb29-7640-4d4d-bd55-7ad212721d9e)
I tried also to compile the code at page 12, maybe the idea here was to show the argument in implicit template instantiation, therefore something like `f<int>(3);`?
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/172ebb41-3ed2-4537-88d9-c510f40d1baf)
And yes, I am Italian too, I am from Turin :smile:
Grazie mille e tanti auguri di Buon Natale!
Ciao
Gianmario
